src/foam/lib/json/Outputter.java

```diff
@@ -111,15 +111,42 @@ public void outputString(String s) {
   }
 
   public String escape(String s) {
-    return s.replace("\\", "\\\\")
+    s = s.replace("\\", "\\\\")
             .replace("\"", "\\\"")
             .replace("\t", "\\t")
             .replace("\r","\\r")
             .replace("\n","\\n");
+    s = escapeControlCharacters(s);
+    return s;
   }
 
   public String escapeMultiline(String s) {
-    return s.replace("\\", "\\\\");
+    s = s.replace("\\", "\\\\");
+    s = escapeControlCharacters(s);
+    return s;
+  }
+
+  public String escapeControlCharacters(String s) {
+    int lastStart = 0;
+    String escapedString = "";
+    for ( int i=0; i < s.length(); i++ ) {
+      if ( s.charAt(i) >= ' ' ) continue;
```

Comment:
Better do var c = s.charAt(i) and reuse for this line and next since it creates a new object which needs to be garbage collected.